issue installing

Hi Kohhee. I was trying to install your npm package and it was failing during the following install process. I am also rather new on Docusaurus and pdf translations, so thanks in advanced:

make: *** [Release/obj.target/hummus/src/PDFDictionaryDriver.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: make failed with exit code: 2
gyp ERR! stack at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack at ChildProcess.emit (events.js:321:20)
gyp ERR! stack at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Darwin 19.2.0
gyp ERR! command "/usr/local/Cellar/node/13.8.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/usr/local/lib/node_modules/hummus/binding/hummus.node" "--module_name=hummus" "--module_path=/usr/local/lib/node_modules/hummus/binding" "--napi_version=5" "--node_abi_napi=napi"
gyp ERR! cwd /usr/local/lib/node_modules/hummus
gyp ERR! node -v v13.8.0
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/Cellar/node/13.8.0/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/usr/local/lib/node_modules/hummus/binding/hummus.node --module_name=hummus --module_path=/usr/local/lib/node_modules/hummus/binding --napi_version=5 --node_abi_napi=napi' (1)
node-pre-gyp ERR! stack at ChildProcess. (/usr/local/lib/node_modules/hummus/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack at ChildProcess.emit (events.js:321:20)
node-pre-gyp ERR! stack at maybeClose (internal/child_process.js:1026:16)
node-pre-gyp ERR! stack at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
node-pre-gyp ERR! System Darwin 19.2.0
node-pre-gyp ERR! command "/usr/local/Cellar/node/13.8.0/bin/node" "/usr/local/lib/node_modules/hummus/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /usr/local/lib/node_modules/hummus
node-pre-gyp ERR! node -v v13.8.0
node-pre-gyp ERR! node-pre-gyp -v v0.10.3
node-pre-gyp ERR! not ok
Failed to execute '/usr/local/Cellar/node/13.8.0/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/usr/local/lib/node_modules/hummus/binding/hummus.node --module_name=hummus --module_path=/usr/local/lib/node_modules/hummus/binding --napi_version=5 --node_abi_napi=napi' (1)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! hummus@1.0.108 install: node-pre-gyp install --fallback-to-build $EXTRA_NODE_PRE_GYP_FLAGS
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the hummus@1.0.108 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

--- a/package.json
+++ b/package.json
@@ -25,8 +25,8 @@
     "chalk": "^3.0.0",
     "commander": "^4.1.1",
     "express": "^4.17.1",
-    "hummus": "^1.0.108",
     "memory-streams": "^0.1.3",
+    "pdf-lib": "^1.7.0",
     "puppeteer": "^2.1.1"
   },
   "files": [

--- a/src/index.ts
+++ b/src/index.ts
@@ -2,23 +2,23 @@ import chalk = require('chalk');
 import puppeteer = require('puppeteer');
 import express = require('express');
 import { AddressInfo } from 'net';
+import { PDFDocument } from 'pdf-lib';
 
-const { PDFRStreamForBuffer, createWriterToModify, PDFStreamForResponse } = require('hummus');
 const { WritableStream } = require('memory-streams');
 const fs = require('fs');
 let generatedPdfBuffers: Array<Buffer> = [];
 
-const mergePdfBuffers = (pdfBuffers: Array<Buffer>) => {
-  const outStream = new WritableStream();
-  const [firstPdfRStream, ...restPdfRStreams] = pdfBuffers.map(pdfBuffer => new PDFRStreamForBuffer(pdfBuffer));
-  const pdfWriter = createWriterToModify(firstPdfRStream, new PDFStreamForResponse(outStream));
+async function mergePdfBuffers(pdfBuffers: Array<Buffer>) {
+  const outDoc = await PDFDocument.create();
+  for (const pdfBuffer of pdfBuffers) {
+    const docToAdd = await PDFDocument.load(pdfBuffer);
+    const pages = await outDoc.copyPages(docToAdd, docToAdd.getPageIndices());
+    for (const page of pages) {
+      outDoc.addPage(page);
+    }
+  }
 
-  restPdfRStreams.forEach(pdfRStream => pdfWriter.appendPDFPagesFromPDF(pdfRStream));
-
-  pdfWriter.end();
-  outStream.end();
-
-  return outStream.toBuffer();
+  return outDoc.save();
 };
 
 const getURL = (origin: string, filePath: string) => {
@@ -127,7 +127,7 @@ export async function generatePdf(
   }
   await browser.close();
 
-  const mergedPdfBuffer = mergePdfBuffers(generatedPdfBuffers);
+  const mergedPdfBuffer = await mergePdfBuffers(generatedPdfBuffers);
   fs.writeFileSync(`${filename}`, mergedPdfBuffer);
 }
 

